core/src/DevStep.cpp

```diff
@@ -7,8 +7,22 @@
 
 #include "include/DevStep.hpp"
 
+#include "include/DiagnosticOutputModule.hpp"
+
 namespace Nextsim {
 
-void DevStep::iterate(const TimestepTime& tst) { pData->update(tst); }
+void DevStep::init()
+{
+    Module::setImplementation<IDiagnosticOutput>("Nextsim::SimpleOutput");
+    Module::getImplementation<IDiagnosticOutput>().setFilenamePrefix("diagnostic");
+}
+
+void DevStep::iterate(const TimestepTime& tst)
+{
+    pData->update(tst);
+    // XIOS wants all the fields, every timestep, so I guess that's what everyone gets
+    ModelState overallState = pData->getStateRecursive(true);
+    Module::getImplementation<IDiagnosticOutput>().outputState(overallState, tst);
+}
```

Comment:
In every time step we will have IO? Maybe this should be made optional, e.g. output every X time steps, output only at the end or every time step?

Reply:
Currently yes. My understanding of XIOS is that it takes all the data every timestep and decides what to do with it. For the same of simplicity, this basic output takes the same fields (all of them, all the time) and immediately dumps them to a netCDF file. XIOS (presumably) has some control of what fields are taken when, and when to actually create files. At this point, I don't this I should put effort into duplicating a more sophisticated control of the output. Unless there is a consensus that a more useful and controlled output system is required in the interim.